ZenPacks/zenoss/LinuxMonitor/FileSystem.py

```diff
@@ -86,19 +86,24 @@ def impacting_object(self):
 
         return self.device()
 
-    @FunctionCache("LinuxFileSystem_getRRDTemplateName", default_timeout=60)
-    def getRRDTemplateName(self):
+    def getRRDTemplates(self):
         """
-        Return name of monitoring template to bind to this component.
-
-        Return non-existent template name if this FileSystem has external storage
-        to prevent monitoring for such component on Linux device or
-        returns the name of an appropriate template for this FileSystem otherwise.
+        For NFS mounts replace general FileSystem template with NFS-specific one.
         """
-        storage_server = list(self.getStorageServers())
-        if storage_server:
-            return "FileSystem_NFS_Client"
-        return "FileSystem"
+        old_templates = super(FileSystem, self).getRRDTemplates()
+
+        if self.type != 'nfs':
+            return old_templates
+
+        new_templates = []
+
+        for template in old_templates:
+            if 'FileSystem' in template.id and 'NFS_Client' not in template.id:
+                new_templates.append(self.getRRDTemplateByName('FileSystem_NFS_Client'))
```

Comment:
Are you attempting to deal with the possible _-replacement_ and _-addition_ custom templates used by zenpacklib here? If so, something like the following would be more comprehensive.

```python
new_templates.append(
    self.getRRDTemplateByName(
        template.id.replace("FileSystem", "FileSystem_NFS_Client")))
```